File: Data Structures & Algorithms/products-of-array-discluding-self/submission-9.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        zerocount = 0 
        tempall = 1
        for n in nums:
            if n != 0:
                tempall *= n
            else:
                zerocount += 1
        if zerocount>1 : 
            return [0] * len(nums)
        res = [0] * len(nums) 
        for i in range(len(nums)):
            if zerocount>0:
                if nums[i] == 0:
                    res[i] = tempall
                else:
                    res[i] = 0
            else:
                res[i] = tempall // nums[i]
        return res;


        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-9.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> pairs;
        vector<vector<int>> buckets(nums.size()+1);
        for(int n :nums)
        {
            pairs[n]++;
        }
        for(auto & pair : pairs)
        {
            buckets[pair.second].push_back(pair.first);
        }
        vector<int> res;
        for(int i = buckets.size() - 1 ; i>= 0 ; i--)
        {
            for(int num : buckets[i])
            {
                if(res.size() == k)
                {
                    return res;
                }
                res.push_back(num);
            }
        }
        return res;
        
    }
};
